integration/tests/access/rest_state_stream_test.go

```diff
@@ -0,0 +1,230 @@
+		testnet.WithAdditionalFlag(fmt.Sprintf("--required-construction-seal-approvals=%d", 1)),
+		testnet.WithLogLevel(zerolog.FatalLevel),
+	}
+
+	nodeConfigs := []testnet.NodeConfig{
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleVerification, testnet.WithLogLevel(zerolog.FatalLevel)),
+		bridgeANConfig,
+		ghostNode,
+	}
+
+	conf := testnet.NewNetworkConfig("access_api_test", nodeConfigs)
+	s.net = testnet.PrepareFlowNetwork(s.T(), conf, flow.Localnet)
+
+	// start the network
+	s.T().Logf("starting flow network with docker containers")
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	s.net.Start(s.ctx)
+}
+
+// TestRestEventStreaming tests event streaming route on REST
+func (s *RestStateStreamSuite) TestRestEventStreaming() {
+	restAddr := s.net.ContainerByName(testnet.PrimaryAN).Addr(testnet.RESTPort)
+
+	s.T().Run("subscribe events", func(t *testing.T) {
+		startBlockId := flow.ZeroID
+		startHeight := uint64(0)
+		url := getSubscribeEventsRequest(restAddr, startBlockId, startHeight, nil, nil, nil)
+
+		client, err := getWSClient(s.ctx, url)
+		require.NoError(t, err)
+
+		var receivedEventsResponse []*state_stream.EventsResponse
+
+		go func() {
+			time.Sleep(10 * time.Second)
+			// close connection after 10 seconds
+			client.Close()
+		}()
+
+		eventChan := make(chan *state_stream.EventsResponse)
+		go func() {
+			for {
+				resp := &state_stream.EventsResponse{}
+				err := client.ReadJSON(resp)
+				if err != nil {
+					if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway) {
+						s.T().Logf("unexpected close error: %v", err)
+						require.NoError(s.T(), err)
+					}
+					close(eventChan) // Close the event channel when the client connection is closed
+					return
+				}
+				eventChan <- resp
+			}
+		}()
+
+		// collect received events during 10 seconds
+		for eventResponse := range eventChan {
+			receivedEventsResponse = append(receivedEventsResponse, eventResponse)
+		}
+
+		// check events
+		s.requireEvents(receivedEventsResponse)
+	})
+}
+
+// requireEvents is a helper function that encapsulates logic for comparing received events from rest state streaming and
+// events which received from grpc api
+func (s *RestStateStreamSuite) requireEvents(receivedEventsResponse []*state_stream.EventsResponse) {
+	// make sure there are received events
+	require.GreaterOrEqual(s.T(), len(receivedEventsResponse), 1, "expect received events")
```

Comment:
this will pass as long as at least one response is returned, regardless of whether or not there were any errors. We should also require that there is at least one response with a non-empty set of events. 

Maybe you could keep a count in the inner `receivedEventMap` loop below, then assert at the end that the count is > 0?